### PR TITLE
Interactions in converters

### PR DESCRIPTION
**Pull request by xXenvy**
## Summary

Pull request adding interaction support to converters

## Checklist

- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/ext/commands/converter.py [hunk 1]
```diff
@@ -284,17 +288,21 @@ class UserConverter(IDConverter[nextcord.User]):
...
         state = ctx._state
 
+        if isinstance(ctx, nextcord.Interaction):
+            bot = ctx.client
+        else:
+            bot = ctx.bot
         if match is not None:
             user_id = int(match.group(1))
-            result = ctx.bot.get_user(user_id) or _utils_get(ctx.message.mentions, id=user_id)
+            result = bot.get_user(user_id) or _utils_get(ctx.message.mentions, id=user_id)
```
Comment (ooliver1) [inline nextcord/ext/commands/converter.py:302]: `ctx.message` in interactions is different to `Context`, and may be `None`.

**Comment by EmreTech** [comment]
Why is this needed? Application Commands have their own converters.

**Review by ooliver1 — CHANGES_REQUESTED** [review]
These might do well with a [`.. versionchanged`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged).

**Comment by ooliver1** [comment]
It might be better to utilise `OptionConverter` since that is *designed* for application commands, instead of retroactively applying to `ext.commands`.

**Comment by ooliver1** [comment]
>  [x] If code changes were made then they have been tested. 

I'm also not too sure if you did in fact test this, since the pyright issues show this will not work in some cases.

**Comment by EmreTech** [comment]
`OptionConverter` already takes in an interaction argument though, in its `convert` method.

**Comment by ooliver1** [comment]
> `OptionConverter` already takes in an interaction argument though, in its `convert` method.

As in, providing default converters.

**Pull request closed without merging**